staging/src/k8s.io/apiserver/pkg/util/proxy/streamtranslator.go

```diff
@@ -0,0 +1,149 @@
+type StreamTranslatorHandler struct {
+	// Location is the location of the upstream proxy. It is used as the location to Dial on the upstream server
+	// for upgrade requests.
+	Location *url.URL
+	// Transport provides an optional round tripper to use to proxy. If nil, the default proxy transport is used
+	Transport http.RoundTripper
+	// Responder is passed errors that occur while setting up proxying.
+	Responder apimachineryproxy.ErrorResponder
+	// Options define the requested streams (e.g. stdin, stdout).
+	Options Options
+}
+
+// NewStreamTranslatorHandler creates a new proxy handler. Responder is required for returning
+// errors to the caller.
+func NewStreamTranslatorHandler(location *url.URL, transport http.RoundTripper, responder apimachineryproxy.ErrorResponder, opts Options) *StreamTranslatorHandler {
+	return &StreamTranslatorHandler{
+		Location:  location,
+		Transport: transport,
+		Responder: responder,
+		Options:   opts,
+	}
+}
+
+func (h *StreamTranslatorHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	// Create WebSocket server, including particular streams requested.
+	websocketStreams, err := webSocketServerStreams(req, w, h.Options)
+	if err != nil {
+		klog.Errorf("websocket createStreams error: %v", err)
+		h.Responder.Error(w, req, err)
+		return
+	}
+	defer websocketStreams.conn.Close()
+
+	// Create the SPDY client connection proxied through kubelet.
+	tlsConfig, err := utilnet.TLSClientConfig(h.Transport)
+	if tlsConfig != nil && err == nil {
+		// Manually set NextProtos to "http/1.1", since http/2.0 will NOT upgrade a connection.
+		tlsConfig.NextProtos = []string{"http/1.1"}
+	} else {
+		klog.Infof("Unable to unwrap transport %T to get at TLS config: %v", h.Transport, err)
+	}
+	spdyRoundTripper := spdy.NewRoundTripper(tlsConfig)
+	spdyExecutor, err := remotecommand.NewSPDYExecutorForTransports(spdyRoundTripper, spdyRoundTripper, "POST", h.Location)
+	if err != nil {
+		klog.Errorf("websocket createStreams error: %v", err)
+		h.Responder.Error(w, req, err)
+		return
+	}
+
+	// Wire the WebSocket server streams output to the SPDY client input.
+	opts := remotecommand.StreamOptions{}
+	if h.Options.Stdin {
+		opts.Stdin = websocketStreams.stdinStream
+	}
+	if h.Options.Stdout {
+		opts.Stdout = websocketStreams.stdoutStream
+	}
+	if h.Options.Stderr {
+		opts.Stderr = websocketStreams.stderrStream
+	}
+	if h.Options.Tty {
+		opts.Tty = true
+		opts.TerminalSizeQueue = &translatorSizeQueue{resizeChan: websocketStreams.resizeChan}
+	}
+	// Start the SPDY client with connected streams. Output from the WebSocket server
+	// streams will be forwarded into the SPDY client.
+	err = spdyExecutor.StreamWithContext(req.Context(), opts)
+	if err != nil {
+		klog.Errorf("spdyExecutor StreamWithContext() error: %v", err)
+		// Report error back to the WebSocket client.
+		//nolint:errcheck
+		if exitErr, ok := err.(exec.CodeExitError); ok && exitErr.Exited() {
```

Comment:
curious why lint flagged this... should we be doing a more resilient `errors.As` check instead?

Reply:
lint flagged the two instances of `websocketStreams.writeStatus` within the block, which return an error. The returned error is ignored because it's an error while we're attempting to return an error.